app/models/transfer.rb

```diff
@@ -59,11 +61,35 @@ def storage_locations_belong_to_organization
     return if organization.nil?
 
     unless organization.storage_locations.include?(from)
-      errors.add :from, "from location must belong to organization"
+      errors.add :from, "location must belong to organization"
     end
 
     unless organization.storage_locations.include?(to)
-      errors.add :to, "to location must belong to organization"
+      errors.add :to, "location must belong to organization"
+    end
+  end
+
+  def storage_locations_must_be_different
+    return if organization.nil? || to_id.nil?
+
+    if from_id == to_id
+      errors.add :to, "location must be different than from location"
+    end
+  end
+
+  def from_storage_locations_must_have_enough_to_transfer_out
+    return if organization.nil? || from.nil?
```

Comment:
I'm not sure if this code/validation is necessary. Can't we just rely on the transaction to tell us if there's insufficient inventory and then rollback accordingly?

Reply:
Possibly? 

Honestly I don't remember what necessitated creating this (it's been awhile!) but you could be right.

I'll take a look at this tomorrow.